File: kt/main/kotlin/moduleconfig/pullwithpushoverride/fs/Baz.kt
package tryout.moduleconfig.pullwithpushoverride.fs

import tryout.moduleconfig.pullwithpushoverride.config.makeCfgSrc

data class BazCfgInfo(
	val w: String
)

var bazCfgSrc = makeCfgSrc<BazCfgInfo>(null)

fun baz() {
	println(bazCfgSrc().w.length);
}

File: rs/pullwithpushoverride/src/fs/baz.rs
use crate::config::cfg_src::CfgSrc;
use arc_swap::ArcSwap;
use once_cell::sync::Lazy;

#[derive(Debug)]
pub struct BazCfgInfo {
    pub w: String,
}

pub static bazCfgSrc: Lazy<ArcSwap<CfgSrc<BazCfgInfo>>> =
    Lazy::new(|| ArcSwap::from_pointee(CfgSrc::<BazCfgInfo>::nil()));

pub fn baz() {
    println!("bazCfgSrc().w.length(): {}", bazCfgSrc.load().get().w.len());
}
